common/lib/session_state_service_impl.ts

```diff
@@ -61,7 +61,9 @@ export class SessionStateServiceImpl implements SessionStateService {
       this.sessionState.autoCommit.resetPristineValue();
       this.setupPristineAutoCommit();
       try {
-        await newClient.setAutoCommit(this.sessionState.autoCommit.value);
+        await newClient.targetClient.query(this.pluginService.getDialect().getSetAutoCommitQuery(this.sessionState.autoCommit.value));
+        this.setAutoCommit(this.sessionState.autoCommit.value);
```

Comment:
The implementation of `setAutoCommit` is like the following
```ts
this.sessionState.autoCommit.value = autoCommit;
```
with some additional logging, so this line is essentially doing
```ts
this.sessionState.autoCommit.value = this.sessionState.autoCommit.value;
```
Curious if this call is necessary? Same comment for the other `this.set....` calls in this method.

Reply:
Can we add getters and setters to the `SessionState` class so we don't have to keep calling `this.sessionState.xxx.value`
```ts
this.sessionState.readOnly = readOnly
```